homeassistant/components/numato/binary_sensor.py

```diff
@@ -0,0 +1,109 @@
+"""
+Binary sensor platform integration for Numato USB GPIO expanders.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/integrations/numato#binary-sensor
+"""
+import logging
+
+from numato_gpio import NumatoGpioError
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import PLATFORM_SCHEMA, BinarySensorDevice
+import homeassistant.components.numato as numato
+from homeassistant.const import CONF_ID, DEVICE_DEFAULT_NAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_INVERT_LOGIC = "invert_logic"
+CONF_PORTS = "ports"
+CONF_DEVICES = "devices"
+CONF_DEVICE_ID = "id"
+DEFAULT_INVERT_LOGIC = False
+
+DEPENDENCIES = ["numato"]
+
+_PORTS_SCHEMA = vol.Schema({cv.positive_int: cv.string})
+_DEVICE_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_ID): cv.positive_int,
+        vol.Required(CONF_PORTS): _PORTS_SCHEMA,
+        vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    }
+)
+_DEVICES_SCHEMA = vol.All(list, [_DEVICE_SCHEMA])
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({vol.Required(CONF_DEVICES): _DEVICES_SCHEMA})
```

Comment:
We no longer allow integrations to use a platform config section. See https://github.com/home-assistant/architecture/blob/master/adr/0007-integration-config-yaml-structure.md.

Please refactor this integration to have the whole config schema in the base integration module and the config yaml section under the top level integration domain key.

Reply:
@MartinHjelmare Ouch! Though I can really understand that kind of KISS decision, it hurts a little. So close....
Do you by any chance know a good example integration which is already designed like this and has some resemblance to the numato config (like multiple devices, entities from different platforms, sync API, ...). It would speed up the refactoring.

Reply:
The octoprint integration has that. It's not up to standard in all regards though, as it lacks a standalone library for its requests.

Reply:
Okay, thanks, got it. But quick question: Is it okay to store some of the config in hass.data[DOMAIN] to make it available during platform (e.g. binary_sensor) setup?

Reply:
Yes, that's ok.

Reply:
Should be fixed by now. Please consider to continue the review. Thank you!